tests/Database/DatabaseEloquentHasOneThroughIntegrationTest.php

```diff
@@ -0,0 +1,489 @@
+<?php
+
+namespace Illuminate\Tests\Database;
+
+use PHPUnit\Framework\TestCase;
+use Illuminate\Database\Eloquent\SoftDeletes;
+use Illuminate\Database\Capsule\Manager as DB;
+use Illuminate\Database\Eloquent\Model as Eloquent;
+
+class DatabaseEloquentHasOneThroughIntegrationTest extends TestCase
+{
+    public function setUp()
+    {
+        $db = new DB;
+
+        $db->addConnection([
+            'driver'    => 'sqlite',
+            'database'  => ':memory:',
+        ]);
+
+        $db->bootEloquent();
+        $db->setAsGlobal();
+
+        $this->createSchema();
+    }
+
+    /**
+     * Setup the database schema.
+     *
+     * @return void
+     */
+    public function createSchema()
+    {
+        $this->schema()->create('users', function ($table) {
+            $table->increments('id');
+            $table->string('email')->unique();
+            $table->unsignedInteger('position_id')->unique()->nullable();
+            $table->string('position_short');
+            $table->timestamps();
+            $table->softDeletes();
+        });
+
+        $this->schema()->create('contracts', function ($table) {
+            $table->increments('id');
+            $table->integer('user_id')->unique();
+            $table->string('title');
+            $table->text('body');
+            $table->string('email');
+            $table->timestamps();
+        });
+
+        $this->schema()->create('positions', function ($table) {
+            $table->increments('id');
+            $table->string('name');
+            $table->string('shortname');
+            $table->timestamps();
+        });
+    }
+
+    /**
+     * Tear down the database schema.
+     *
+     * @return void
+     */
+    public function tearDown()
```

Comment:
Going to stick to this for now since the other tests have it as well. I see what you're saying but it's best to stick to the convention of having them if all the other tests do as well.